client/src/contexts/NewForm.jsx

```diff
@@ -1,25 +1,99 @@
-import React, { useState } from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useContext, useEffect } from 'react';
+import axios from 'axios';
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
+import Console from '../Console';
+import ReviewInput from '../components/Ratings/Review/ReviewInput';
 import { Modal, ModalParent, ModalClose } from './Shared.styled';
-// import Console from '../../Console';
+import { useMeta } from './ReviewMeta';
+import { ProductIDContext } from './ProductIDContext';
 
-// formtype: review, question
-export default function NewForm({ formtype, productName, showModal }) {
+export default function NewForm({
+  formtype, productName, showModal, setShowModal, ...args
+}) {
   let type;
-  const [modalStatus, setModalStatus] = useState(true);
-  // const [data, setData] = useState({});
-  // const handFormChange = () =
-  const review = {
+  let bodyNote;
+  const meta = useMeta();
+  const factorList = (meta) ? meta.characteristics : {};
+  // CONFIRM WITH ALEX this default won't imapct his section
+  const [data, setData] = useState({ recommendation: 'true' });
+  const productId = useContext(ProductIDContext);
+  const handleChange = (e) => setData((prevState) => (
+    { ...prevState, [e.target.name]: e.target.value }));
+
+  useEffect(() => {
+    const espExit = (e) => {
+      if (e.key === 'Escape') {
+        setShowModal(false);
+      }
+    };
+    window.addEventListener('keydown', espExit);
+    return () => window.removeEventListener('keydown', espExit);
+  }, []);
+
+  const handleOnSubmit = (e) => {
```

Comment:
Your code looks great, so I'm just splitting hairs at this point. This is a really long function that does a bunch of things when it submits. A future refactor could be to split it up into its discrete parts so that this event handler would just read like:

```
if (formtype === 'reviews') {
  const data = formatData();
  sendRequest(data);
}
```

But honestly who really cares - it does what it needs to do.

Reply:
That is exactly what I need! Thanks for the tip